packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -266,17 +273,29 @@ const BaseButton = ({
   iconPosition = 'left',
   isDisabled = false,
   isFullWidth = false,
+  isLoading = false,
   onClick,
   type = 'button',
   children,
+  accessibilityLabel,
 }: BaseButtonProps): ReactElement => {
+  const disabled = isLoading || isDisabled;
   const { theme, platform } = useTheme();
   const buttonTypographyTokens = buttonTypography[platform];
   if (!Icon && !children?.trim()) {
     throw new Error(
       `[Blade: BaseButton]: At least one of icon or text is required to render a button.`,
     );
   }
+
+  const prevLoading = usePrevious(isLoading);
```

Comment:
usePrevious will return the previous state which is being passed to it. 

if isLoading is false in the latest render, it will return true (the previous render's value)

We want this to be able to detect when the loading is stopped 
```ts
    if (!isLoading && prevLoading) announce('Stopped loading');

```

Reply:
Discussed, makes sense.